caddyhttp/gzip/responsefilter.go

```diff
@@ -25,6 +25,14 @@ func (l LengthFilter) ShouldCompress(w http.ResponseWriter) bool {
 	return l != 0 && int64(l) <= length
 }
 
+// SkipGzipped is ResponseFilter that will discard already gzipped files
```

Comment:
Technically, it will skip compressing responses that have a non-empty Content-Encoding header; it could be any value...